lightning/src/offers/static_invoice.rs

```diff
@@ -0,0 +1,1173 @@
+		assert_eq!(invoice.offer_features(), &OfferFeatures::empty());
+		assert_eq!(invoice.absolute_expiry(), None);
+		assert_eq!(invoice.offer_message_paths(), &[blinded_path()]);
+		assert_eq!(invoice.message_paths(), &[blinded_path()]);
+		assert_eq!(invoice.issuer(), None);
+		assert_eq!(invoice.supported_quantity(), Quantity::One);
+		assert_ne!(invoice.signing_pubkey(), recipient_pubkey());
+		assert_eq!(invoice.chain(), ChainHash::using_genesis_block(Network::Bitcoin));
+		assert_eq!(invoice.payment_paths(), payment_paths.as_slice());
+		assert_eq!(invoice.created_at(), now);
+		assert_eq!(invoice.relative_expiry(), DEFAULT_RELATIVE_EXPIRY);
+		#[cfg(feature = "std")]
+		assert!(!invoice.is_expired());
+		assert!(invoice.fallbacks().is_empty());
+		assert_eq!(invoice.invoice_features(), &Bolt12InvoiceFeatures::empty());
+
+		let offer_signing_pubkey = offer.signing_pubkey().unwrap();
+		let message = TaggedHash::from_valid_tlv_stream_bytes(SIGNATURE_TAG, &invoice.bytes);
+		assert!(
+			merkle::verify_signature(&invoice.signature, &message, offer_signing_pubkey).is_ok()
+		);
+
+		let paths = vec![blinded_path()];
+		let metadata = vec![42; 16];
+		assert_eq!(
+			invoice.as_tlv_stream(),
+			(
+				OfferTlvStreamRef {
+					chains: None,
+					metadata: Some(&metadata),
+					currency: None,
+					amount: None,
+					description: None,
+					features: None,
+					absolute_expiry: None,
+					paths: Some(&paths),
+					issuer: None,
+					quantity_max: None,
+					node_id: Some(&offer_signing_pubkey),
+				},
+				InvoiceTlvStreamRef {
+					paths: Some(Iterable(payment_paths.iter().map(|(_, path)| path))),
+					blindedpay: Some(Iterable(payment_paths.iter().map(|(payinfo, _)| payinfo))),
+					created_at: Some(now.as_secs()),
+					relative_expiry: None,
+					payment_hash: None,
+					amount: None,
+					fallbacks: None,
+					features: None,
+					node_id: Some(&offer_signing_pubkey),
+					message_paths: Some(&paths),
+				},
+				SignatureTlvStreamRef { signature: Some(&invoice.signature()) },
+			)
+		);
+
+		if let Err(e) = StaticInvoice::try_from(buffer) {
+			panic!("error parsing invoice: {:?}", e);
+		}
+	}
+
+	#[cfg(feature = "std")]
+	#[test]
+	fn builds_invoice_from_offer_with_expiration() {
+		let node_id = recipient_pubkey();
+		let now = now();
+		let expanded_key = ExpandedKey::new(&KeyMaterial([42; 32]));
+		let entropy = FixedEntropy {};
+		let secp_ctx = Secp256k1::new();
+
+		let future_expiry = Duration::from_secs(u64::max_value());
+		let past_expiry = Duration::from_secs(0);
+
+		let valid_offer =
+			OfferBuilder::deriving_signing_pubkey(node_id, &expanded_key, &entropy, &secp_ctx)
+				.path(blinded_path())
+				.absolute_expiry(future_expiry)
+				.build()
+				.unwrap();
+
+		let invoice = StaticInvoiceBuilder::for_offer_using_keys(
+			&valid_offer,
+			payment_paths(),
+			vec![blinded_path()],
+			now,
+			&expanded_key,
+			&secp_ctx,
+		)
+		.unwrap()
+		.build_and_sign(&secp_ctx)
+		.unwrap();
```

Comment:
Indent? Likewise throughout.

Reply:
I'm using rustfmt 😀

Reply:
Huh, strange that it's inconsistent with when the first call can fit on one line.